infra_tooling_account/infra_tooling_account/infra_tooling_grafana_stack.py

```diff
@@ -0,0 +1,383 @@
+                    ),
+                ),
+            ],
+            destination_bucket=settings_bucket,
+            destination_key_prefix="settings/grafana/conf",
+        )
+
+    def create_grafana_instance(
+        self,
+        grafana_vpc_id: str,
+        grafana_security_group_id: str,
+        input_timestream_kms_key_arn: str,
+        input_timestream_database_arn: str,
+        input_settings_bucket_arn: str,
+        grafana_admin_secret_arn: str,
+        grafana_instance_type: str,
+        grafana_bitnami_image: str,
+        grafana_key_pair_name: str,
+        settings_bucket_name: str,
+        grafana_admin_secret_name: str,
+    ) -> ec2.Instance:
+        """Creates Grafana instance with the provisioned dashboards.
+
+        Args:
+            grafana_vpc_id (str): VPC ID for Grafana instance
+            grafana_security_group_id (str): Security Group ID for Grafana instance
+            input_timestream_kms_key_arn (str): ARN of Timestream KMS Key
+            input_timestream_database_arn (str): ARN of Timestream DB
+            input_settings_bucket_arn (str): ARN of Settings S3 Bucket
+            grafana_admin_secret_arn (str): ARN of Grafana secret
+            grafana_instance_type (str): Grafana Instance type
+            grafana_image (str): Bitnami Grafana image from AWS Marketplace
+            grafana_key_pair_name (str): Grafana Key Pair name
+            settings_bucket_name (str): Settings S3 Bucket name
+            grafana_admin_secret_name (str): Grafana secret name
+
+        Returns:
+            ec2.Instance: Grafana Instance
+        """
+        # Import the VPC, security group provided
+        vpc = ec2.Vpc.from_lookup(self, "VpcImport", vpc_id=grafana_vpc_id)
+        security_group = ec2.SecurityGroup.from_lookup_by_id(
+            self, "SecurityGroupImport", security_group_id=grafana_security_group_id
+        )
+
+        # Create Grafana Role
+        grafana_role = iam.Role(
+            self,
+            "GrafanaInstanceRole",
+            assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"),
+            description="Role assumed by Grafana instance",
+            role_name=AWSNaming.IAMRole(self, "grafana-instance"),
+        )
+
+        grafana_role.add_to_policy(
+            # to be able to decrypt data from Timestream DB
+            iam.PolicyStatement(
+                actions=["kms:Decrypt"],
+                effect=iam.Effect.ALLOW,
+                resources=[input_timestream_kms_key_arn],
+            )
+        )
+        grafana_role.add_to_policy(
+            # to be able to make Timestream API calls from Grafana instance
+            iam.PolicyStatement(
+                actions=[
+                    "timestream:DescribeEndpoints",
+                    "timestream:ListDatabases",
+                    "timestream:SelectValues",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
```

Comment:
Is it possible to limit the permissions to our specific Timestream DB only?

Reply:
I believe, for DescribeEndpoints action, we should keep "*" - kind of internal limitation from Timestream.
For others - I'm not sure, let's check and tighten permissions is possible

Reply:
Yes, right. When I set the DescribeEndpoints action for my Timstream DB only, I'm getting this error in Grafana:

AccessDeniedException: User: arn:aws:sts::account_id:assumed-role/role-salmon-grafana-instance-devay/i-0a55a4093dc8364e7 is not authorized to perform: timestream:DescribeEndpoints because no identity-based policy allows the timestream:DescribeEndpoints action

Reply:
For others - I'll re-check, thank you!

Reply:
the same error "AccessDeniedException" if the action timestream:SelectValues is set only for my DB
and as for "timestream:ListDatabases", as I can see, the Timestream plugin will not work for new data sources added in Grafana UI, since we will not be able to select the database name while establishing the connection

Reply:
I'm fine with leaving "*" if it's needed, but, I guess there might be more details :), so let's have a separate discussion about this during the next sync-up